core/app/models/spree/price.rb

```diff
@@ -6,13 +6,21 @@ class Price < Spree::Base
 
     belongs_to :variant, -> { with_deleted }, class_name: 'Spree::Variant', touch: true
 
+    after_initialize :valid_from_today!, if: -> { valid_from.blank? }
+
     validate :check_price
     validates :amount, allow_nil: true, numericality: {
       greater_than_or_equal_to: 0,
       less_than_or_equal_to: MAXIMUM_AMOUNT
     }
 
-    after_save :set_default_price
+    scope :valid_before, -> (date) { where("#{self.table_name}.valid_from <= ?", date).order(valid_from: :desc) }
```

Comment:
If you want this scope to be called elsewhere, you need to include `with_deleted` scope